### PR TITLE
test(storage): connection pool size option for benchmarks

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9679)

### CODE REVIEW

File: google/cloud/storage/benchmarks/benchmark_utils.cc [hunk 1]
```diff
@@ -177,9 +177,13 @@ void PrintOptions(std::ostream& os, std::string const& prefix,
     os << "\n# " << prefix
        << " Grpc Endpoint: " << options.get<EndpointOption>();
   }
+  if (options.has<gcs::ConnectionPoolSizeOption>()) {
+    os << "\n# " << prefix << " Connection Pool Size: "
```
Comment (dbolduc) [inline google/cloud/storage/benchmarks/benchmark_utils.cc:181]: can we say "REST Connection Pool Size: "? so it's obvious the option doesn't apply for the gRPC case.

**Review by dbolduc — COMMENTED** [review]
is this option unworthy of changes to `*_options_test.cc` ?

File: google/cloud/storage/benchmarks/benchmark_utils.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/benchmark_utils.cc]: Fixed and added the tests

**Review by scotthart — APPROVED** [review]
> Reviewed 3 of 4 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9679)* status: all files reviewed, 1 unresolved discussion (waiting on @dbolduc)

**Pull request merged**